trollmoves/dispatcher.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2012-2019
+#
+# Author(s):
+#
+#   Martin Raspaud <martin.raspaud@smhi.se>
+#   Panu Lahtinen <panu.lahtinen@fmi.fi>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Dispatcher dispatches(!) files to some destination.
+
+It listens to posttroll messages to find out what to dispatch.
+
+
+Format of the configuration file
+================================
+
+.. highlight:: yaml
+
+Example config::
+
+    client1:
+      host: ftp://ftp.client1.com
```

Comment:
Maybe add one level more here, e.g.:
```yaml
  target_host:
    uri: ftp://ftp.client1.com
    connection_parameters:
      credential_file: .netrc
    filepattern: '{platform_name}_{start_time}.{format}'
    directory: /input_data/{sensor}
```
Just to make it more clear these will are for the target system, not e.g. `filepattern` of the incoming files.

Reply:
so you would like:
```yaml
client1:
  target_host:
    uri: ftp://ftp.client1.com
``` 
etc ?
I'm not sure I like it, it adds an 'unused' level, a level with just one item.

Reply:
Yes, like that. If not, I think there should anyway be a clear separation between the message metadata bits and items relevant to the target host. Oh, if `dispatch` is renamed to `metadata_filters`, and maybe `filepattern` to `target_filepattern` and `directory` to `target_directory` that should do the trick.

Reply:
`dispatch_configs` will be used as per slack conversation